From 4be6c5a090a4f2f19f055aa5f662fd43f2d6a2c9 Mon Sep 17 00:00:00 2001
From: Robin Hack <hack.robin@gmail.com>
Date: Tue, 26 Apr 2016 17:51:46 +0200
Subject: [PATCH 1057/1096] ldb-samba/ldb_matching_rules: Fix CID 1349424 -
 Uninitialized pointer read

Fix unitialized 'visited' value (pointer to pointer) in
ldb_eval_transitive_filter() which passes 'visited' value later to
ldb_eval_transitive_filter_helper().

Signed-off-by: Robin Hack <hack.robin@gmail.com>
Reviewed-by: Jeremy Allison <jra@samba.org>
Reviewed-by: Andreas Schneider <asn@samba.org>

Autobuild-User(master): Jeremy Allison <jra@samba.org>
Autobuild-Date(master): Wed Apr 27 02:12:39 CEST 2016 on sn-devel-144

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11890

(cherry picked from commit f4181f25b4ae3db684e43837449617e75183ecc8)
(cherry picked from commit 5fcfb62d4b0ae750236562bce641baf7e355d5b0)
---
 lib/ldb-samba/ldb_matching_rules.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ldb-samba/ldb_matching_rules.c b/lib/ldb-samba/ldb_matching_rules.c
index 1692a73..637858f 100644
--- a/lib/ldb-samba/ldb_matching_rules.c
+++ b/lib/ldb-samba/ldb_matching_rules.c
@@ -206,7 +206,7 @@ static int ldb_eval_transitive_filter(TALLOC_CTX *mem_ctx,
 	struct dsdb_dn *dn_to_match;
 	const char *dn_oid;
 	unsigned int count;
-	struct dsdb_dn **visited;
+	struct dsdb_dn **visited = NULL;
 
 	schema = dsdb_get_schema(ldb, mem_ctx);
 	if (schema == NULL) {
-- 
2.9.0

